examples/some-issuer/src/bin/discord.rs

```diff
@@ -292,10 +299,26 @@ async fn issue_discord_credential(
         }
     };
 
-    state
-        .issuer
-        .issue_credential(&request.credential, user_id, username)
-        .await
+    let (response_sender, response_receiver) = tokio::sync::oneshot::channel();
+    let data = IssueChannelData {
+        credential: request.credential,
+        user_id,
+        username,
+        response_sender,
+    };
+    if state.issuer_sender.send(data).await.is_err() {
+        tracing::error!("Failed enqueueing transaction. The transaction sender task died.");
+        return Err(StatusCode::INTERNAL_SERVER_ERROR);
+    }
+    if let Ok(r) = response_receiver.await {
+        r.map(Json)
+    } else {
+        // There is no information in the error.
+        tracing::error!(
+            "Failed sending transaction; did not get response from transaction sender."
+        );
+        Err(StatusCode::INTERNAL_SERVER_ERROR)
+    }
```

Comment:
Why not "hide" this implementation in the `lib.rs` and just expose a function for issuing credentials instead? I'm asking because this is how I would have done it, and you probably have a good reason for exposing the channel stuff to the consuming party (remembering a discussion from the implementation of the custom indexer)?